src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs

```diff
@@ -150,6 +151,21 @@ public DatabaseDocumentActions(DocumentDatabase database, BuildVersionType build
 
             public void WriteDocument(DocumentItem item, SmugglerProgressBase.CountsWithLastEtag progress)
             {
+                if (item.Document.NonPersistentFlags.Contain(NonPersistentDocumentFlags.FromSmuggler) &&
+                     ((item.Document.Flags & DocumentFlags.HasRevisions) != 0) &&
+                    ((item.Document.Flags & DocumentFlags.Revision) !=0) || 
+                     ((item.Document.Flags & DocumentFlags.DeleteRevision) != 0))
+                {
+                    using (_database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+                    using (context.OpenReadTransaction())
+                    {
+                        if (_database.DocumentsStorage.Get(context, item.Document.Id) == null)
+                        {
+                            throw new DocumentDoesNotExistException();
```

Comment:
Also, you _must_ provide more details here. For example, what is the missing document?

Reply:
There is also a second parameter there, you can add a detailed message. We should add that to the progress so user will see what is the source of the error maybe? I think we have sth like `result.AddError` there available.